src/vision/rust_image.rs

```diff
@@ -0,0 +1,180 @@
+use std::convert::TryFrom;
+
+use image::{
+    DynamicImage, EncodableLayout, GrayAlphaImage, GrayImage, Rgb32FImage, RgbImage, Rgba32FImage,
+    RgbaImage,
+};
+
+use crate::vision::image::{chw_to_hwc, hwc_to_chw};
+use crate::{Kind, TchError, Tensor};
+
+impl<'i> From<&'i DynamicImage> for Tensor {
```

Comment:
As this is fallible, it might be better to use try-from and let the end user unwrap and handle the error if they want. The same applies to the other `From` impl below.

Reply:
Ok, now all methods use `f_*` version and raise error of `TchError`